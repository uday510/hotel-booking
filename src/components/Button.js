import React from 'react';

const Button = ({ onClick, label, bgColor, textColor }) => {
  const buttonClassName = `p-4 my-4 bg-${bgColor} text-${textColor} w-full rounded-lg`;

  return (
    <button
      className={buttonClassName}
      onClick={onClick}
    >
      {label}
    </button>
  );
};

export default Button;
