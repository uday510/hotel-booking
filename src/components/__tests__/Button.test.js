/**
 * @file Button.test.js
 * @description Test file for the Button component using Jest and react-test-renderer.
 */

import renderer from 'react-test-renderer';
import React from 'react';
import Button from '../Button';

/**
 * Renders a tree using react-test-renderer.
 *
 * @param {ReactNode} tree - The React tree to be rendered.
 * @returns {ReactTestRenderer} The rendered tree.
 */
const renderTree = (tree) => renderer.create(tree);

describe('<Button>', () => {
  /**
   * Snapshot test for rendering the Button component.
   */
  it('should render component', () => {
    expect(renderTree(<Button />).toJSON()).toMatchSnapshot();
  });
});
