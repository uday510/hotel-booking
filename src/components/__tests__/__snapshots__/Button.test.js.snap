// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> should render component 1`] = `
<button
  className="p-4 my-4 bg-undefined text-undefined w-full rounded-lg"
/>
`;
